src/rdyadvance.c

```diff
@@ -244,14 +245,69 @@ PetscErrorCode RDyAdvance(RDy rdy) {
 
   PetscReal interval           = ConvertTimeToSeconds(rdy->config.time.coupling_interval, rdy->config.time.unit);
   PetscReal next_coupling_time = time + interval;
+
+  RDyLogDetail(rdy, "Advancing from t = %g to %g...", ConvertTimeFromSeconds(time, rdy->config.time.unit),
+               ConvertTimeFromSeconds(next_coupling_time, rdy->config.time.unit));
+
+  // if time adaptivity is enabled, try to increase the dt
+  RDyTimeAdaptivitySection *time_adap = &rdy->config.time.adaptivity;
+  if (time_adap->enable) {
+    CourantNumberDiagnostics *cnum_diags = &rdy->courant_num_diags;
+
+    // if previous courant number is valid
+    if (cnum_diags->is_set) {
+      // get current timestep
+      PetscReal dt = rdy->dt;
+
+      if (cnum_diags->max_courant_num < time_adap->target_courant_number) {
+        // timestep can be increased, so find the factor by which timestep can be increased
+        PetscReal factor = PetscMin(time_adap->target_courant_number / cnum_diags->max_courant_num, time_adap->max_increase_factor);
+
+        // increase the timestep
+        dt *= factor;
+
+        // ensure the increase in timestep is less than the coupling timestep
+        if (dt > interval) dt = interval;
+
+        if (rdy->config.logging.level >= LOG_DEBUG) {
+          const char *units  = TimeUnitAsString(rdy->config.time.unit);
+          PetscReal   dt_old = ConvertTimeFromSeconds(rdy->dt, rdy->config.time.unit);
+          PetscReal   dt_new = ConvertTimeFromSeconds(dt, rdy->config.time.unit);
+          RDyLogDebug(rdy, "Increasing dt from %f [%s] to %f [%s]", dt_old, units, dt_new, units);
+        }
+
+        // update the timestep
+        rdy->dt = dt;
+
+      } else {
+        // decrease the timestep
+        PetscReal factor = time_adap->target_courant_number / cnum_diags->max_courant_num;
+
```

Comment:
You can get rid of lines 285-286 and just group both lines of code under one comment.

Reply:
I consolidated more common code outside the if-else loop.